File: js/3.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    var m = {}
    var i = 0, j = 0, ret = 0
    while(j < s.length) {
        if(m[s[j]] === undefined || m[s[j]] < i) {
            ret = Math.max(ret, j - i + 1)
        } else {
            i = m[s[j]] + 1
        }
        m[s[j]] = j ++
    }
    return ret
};

File: go/3.go
func lengthOfLongestSubstring(s string) int {
    m := map[int]int{} 
    i :=0 
    j := 0 
    ret := 0
    for j < len(s) {
        if v, ok := m[int(s[j])]; !ok || v < i {
            if j - i + 1 > ret {
                ret = j - i + 1
            }
        } else {
            i = v + 1
        }
        m[int(s[j])] = j
        j = j + 1
    }
    return ret
}
